lib/foreman_puppet/engine.rb

```diff
@@ -33,7 +33,7 @@ def argument(name, type, *rest, loads: nil, **kwargs, &block)
     engine_name 'foreman_puppet'
     isolate_namespace ForemanPuppet
 
-    config.paths['db/migrate'] << 'db/migrate_foreman' if Gem::Dependency.new('', ">= #{ForemanPuppet::FOREMAN_EXTRACTION_VERSION}").match?('', SETTINGS[:version])
+    config.paths['db/migrate'] << 'db/migrate_foreman' if Gem::Dependency.new('', "> #{ForemanPuppet::FOREMAN_EXTRACTION_VERSION}").match?('', SETTINGS[:version])
```

Comment:
this means that it will be included again when we do 3.0.1, which is not what your intention is, right? you want it to be included on 3.1+, as that version removes the migrations from core?